src/jit/liveness.cpp

```diff
@@ -2225,7 +2225,8 @@ void Compiler::fgComputeLifeLIR(VARSET_TP& life, BasicBlock* block, VARSET_VALAR
                 assert(!node->OperIsLocal());
                 if (!node->IsValue() || node->IsUnusedValue())
                 {
-                    unsigned sideEffects = node->gtFlags & (GTF_SIDE_EFFECT | GTF_SET_FLAGS);
+                    // We are not interested in GTF_CALL, which will be set on nodes with calls in their subtree.
+                    unsigned sideEffects = node->gtFlags & (GTF_ASG | GTF_EXCEPT | GTF_SET_FLAGS);
```

Comment:
For `GTF_EXCEPT` we check that `((sideEffects == GTF_EXCEPT) && !node->OperMayThrow(this))` so it means `remove if the node itself doesn't throw` (We have a checker that guarantee that there are no nodes without `GTF_EXCEPT` which `OperMayThrow`).

but my question is do we care about `GTF_ASG` from subtrees?

Reply:
We probably don't care about `GTF_ASG` from subtrees. What we probably need here is the equivalent of `Compiler::gtUpdateNodeOperSideEffects()` that only returns the side effects of the current node, e.g. `Compiler::getNodeOperSideEffects()`